src/Core/Challenge/Dns/DnspodSolver.php

```diff
@@ -0,0 +1,185 @@
+ */
+class DnspodSolver implements MultipleChallengesSolverInterface, ConfigurableServiceInterface
+{
+    use LoggerAwareTrait;
+    /**
+     * @var DnsDataExtractor
+     */
+    private $extractor;
+
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * @var array
+     */
+    private $cacheZones;
+
+    /**
+     * @var string
+     */
+    private $secretId;
+
+    /**
+     * @var string
+     */
+    private $secretKey;
+
+    /**
+     * @param DnsDataExtractor $extractor
+     * @param ClientInterface  $client
+     */
+    public function __construct(
+        DnsDataExtractor $extractor = null,
+        ClientInterface $client = null
+    ) {
+        $this->extractor = null === $extractor ? new DnsDataExtractor() : $extractor;
+        $this->client = null === $client ? new Client() : $client;
+        $this->logger = new NullLogger();
+    }
+
+    /**
+     * Configure the service with a set of configuration.
+     *
+     * @param array $config
+     */
+    public function configure(array $config)
+    {
+        $this->secretId = $config['secret_id'];
+        $this->secretKey = $config['secret_key'];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(AuthorizationChallenge $authorizationChallenge)
+    {
+        return 'dns-01' === $authorizationChallenge->getType();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function solve(AuthorizationChallenge $authorizationChallenge)
+    {
+        return $this->solveAll([$authorizationChallenge]);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function solveAll(array $authorizationChallenges)
+    {
+        Assert::allIsInstanceOf($authorizationChallenges, AuthorizationChallenge::class);
+
+        $config = array(
+            'SecretId' => $this->secretId,
+            'SecretKey' => $this->secretKey,
+            'RequestMethod' => 'GET',
+            'DefaultRegion' => 'gz'
+        );
+        /**
+         * @var \QcloudApi_Module_Cns
+         */
+        $cns = QcloudApi::load(QcloudApi::MODULE_CNS, $config);
+
+        foreach ($authorizationChallenges as $authorizationChallenge) {
+            $topLevelDomain = $this->getTopLevelDomain($authorizationChallenge->getDomain());
+            $recordName = $this->extractor->getRecordName($authorizationChallenge);
+            $recordValue = $this->extractor->getRecordValue($authorizationChallenge);
+
+            $subDomain = \str_replace('.'.$topLevelDomain.'.', '', $recordName);
+
+            $solve = $cns->RecordCreate([
+                'domain' => $topLevelDomain,
+                'subDomain' => $subDomain,
+                'recordType' => 'TXT',
+                'recordLine' => '默认',
+                'value' => $recordValue,
+            ]);
+
+            if ($solve === false) {
+                throw $cns->getError();
+            }
+
+            $data = json_decode($cns->getLastResponse(), true);
```

Comment:
use `\GuzzleHttp\json_decode` that throws exception in case of invalide response

Reply:
It should be throw, because if ignore wrong response, we can't sure is the dns record added. according to RFC 8555,  auth z can only be requested after sure DNS record add and take affected.

Reply:
I'm thinking about a case where the API don't return Json content.
Maybe a temporary outage that returns a http 200 with a HTML content.

Reply:
> I'm thinking about a case where the API don't return Json content.
> Maybe a temporary outage that returns a http 200 with a HTML content.

If non-200 response or 200 with non-JSON, it will throw, this is no problem because the add-DNS  request was not succeed at the most possible.

Reply:
you can't be sure about that. user can be behind a proxy, that redirect him to a login page for instance.

Reply:
> you can't be sure about that. user can be behind a proxy, that redirect him to a login page for instance.

**You mean an authentication required proxy?**
Yes, that is possible to happen, But i can see not only current actions existed don't concern same situation either.  And also i think it should be deal by ActionInterface or create a AbstractAction to provide a ProxyAuthHandler for guzzle, because the proxy auth progress will be needed by everything restful oriented actions.
 
If anything i was wrong, please correct it, or additional discuss it.